universal-application-tool-0.0.1/app/services/applicant/ApplicantData.java

```diff
@@ -54,16 +47,67 @@ public void setPreferredLocale(Locale locale) {
     this.preferredLocale = locale;
   }
 
+  /**
+   * Checks whether the given path exists in the JSON data. Returns true if the path is present;
+   * false otherwise. Semantically, this checks whether the applicant has answered this question
+   * before.
+   *
+   * @param path the {@link Path} to check
+   * @return true if path is present for this applicant; false otherwise
+   */
+  public boolean hasPath(Path path) {
+    try {
+      this.jsonData.read(path.path());
+    } catch (PathNotFoundException e) {
+      return false;
+    }
+    return true;
+  }
+
+  /**
+   * Returns true if there is any value at the given {@link Path}; false otherwise.
```

Comment:
nit (blocking): "any value" is a little tricky in this context because `null` is a value and it's not clear without reading the code whether or not this can return null. Recommend clarifying that it will return `false` if the value at the path is `null`.

Reply:
Great point, fixed the wording to "non-null"